tests/utils/tensorflow/test_callback.py

```diff
@@ -0,0 +1,71 @@
+from typing import Dict, Text, List
+from pathlib import Path
+
+import pytest
+
+from rasa.core.agent import Agent
+from rasa.core.policies.ted_policy import TEDPolicy
+from rasa.utils.tensorflow.callback import RasaModelCheckpoint
+
+
+@pytest.mark.parametrize(
+    "logs, improved",
```

Comment:
To improve clarity, you could split `logs` into `previous_values` and `new_values`.